test_api.py

```diff
@@ -1871,32 +1874,35 @@ def __iter__(self, *args, **kwargs):
             self._delete_hosts()
             return iterator
 
-    def test_delete_when_one_host_is_deleted(self, emit_event):
+    def test_delete_when_one_host_is_deleted(self):
         host_id = self.added_hosts[0].id
         url = HOST_URL + "/" + host_id
         with patch("api.host.delete_hosts", self.DeleteHostsMock.create_mock([host_id])):
             # One host queried, but deleted by a different process. No event emitted yet returning
             # 200 OK.
-            self.delete(url, 200, return_response_as_json=False)
-            emit_event.assert_not_called()
+            with self.app.app_context():
+                self.delete(url, 200, return_response_as_json=False)
+                self.event_producer.write_event.assert_not_called()
 
-    def test_delete_when_all_hosts_are_deleted(self, emit_event):
+    def test_delete_when_all_hosts_are_deleted(self):
         host_id_list = [self.added_hosts[0].id, self.added_hosts[1].id]
         url = HOST_URL + "/" + ",".join(host_id_list)
         with patch("api.host.delete_hosts", self.DeleteHostsMock.create_mock(host_id_list)):
-            # Two hosts queried, but both deleted by a different process. No event emitted yet
-            # returning 200 OK.
-            self.delete(url, 200, return_response_as_json=False)
-            emit_event.assert_not_called()
+            with self.app.app_context():
+                # Two hosts queried, but both deleted by a different process. No event emitted yet
+                # returning 200 OK.
+                self.delete(url, 200, return_response_as_json=False)
+                self.app.event_producer.write_event.assert_not_called()
 
-    def test_delete_when_some_hosts_is_deleted(self, emit_event):
+    def test_delete_when_some_hosts_is_deleted(self):
         host_id_list = [self.added_hosts[0].id, self.added_hosts[1].id]
         url = HOST_URL + "/" + ",".join(host_id_list)
         with patch("api.host.delete_hosts", self.DeleteHostsMock.create_mock(host_id_list[0:1])):
-            # Two hosts queried, one of them deleted by a different process. Only one event emitted,
-            # returning 200 OK.
-            self.delete(url, 200, return_response_as_json=False)
-            emit_event.assert_called_once()
+            with self.app.app_context():
+                # Two hosts queried, one of them deleted by a different process. Only one event emitted,
+                # returning 200 OK.
+                self.delete(url, 200, return_response_as_json=False)
+                self.app.write_event.assert_called_once()
```

Comment:
Here, we can even test that the event is emitted for the correct host.

```suggestion
                self.assertEqual(host_id_list[1], self.app.event_producer.key)
```